database/reorg_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2015 Conformal Systems LLC.
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package database_test
+
+import (
+	"compress/bzip2"
+	"encoding/binary"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+)
+
+// testReorganization performs reorganization tests for the passed DB type.
+// Much of the setup is copied from the blockchain package, but the test looks
+// to see if each TX in each block in the best chain can be fetched using
+// FetchTxBySha. If not, then there's a bug.
+func testReorganization(t *testing.T, dbType string) {
+	db, teardown, err := createDB(dbType, "reorganization", true)
+	if err != nil {
+		t.Errorf("Failed to create test database (%s) %v", dbType, err)
+		return
+	}
+	defer teardown()
+
+	blocks, err := loadReorgBlocks("reorgblocks.bz2")
+	if err != nil {
+		t.Errorf("Error loading file: %v", err)
+		return
+	}
+
+	for i := int64(0); i <= 2; i++ {
+		blkHash, err := blocks[i].Sha()
+		if err != nil {
+			t.Errorf("Error getting SHA for block %d: %v", i, err)
+			return
+		}
+		_, err = db.InsertBlock(blocks[i])
+		if err != nil {
+			t.Errorf("Error inserting block %d (%v): %v", i, blkHash, err)
+			return
+		}
+		var txIDs []string
+		for _, tx := range blocks[i].Transactions() {
+			txIDs = append(txIDs, tx.Sha().String())
+		}
+	}
+
+	for i := int64(1); i >= 0; i-- {
+		blkHash, err := blocks[i].Sha()
+		if err != nil {
+			t.Errorf("Error getting SHA for block %d: %v", i, err)
+			return
+		}
+		err = db.DropAfterBlockBySha(blkHash)
+		if err != nil {
+			t.Errorf("Error removing block %d for reorganization: %v", i, err)
+			return
+		}
+		// Exercise NewestSha() to make sure DropAfterBlockBySha() updates the
+		// info correctly
+		maxHash, blkHeight, err := db.NewestSha()
+		if err != nil {
+			t.Errorf("Error getting newest block info")
+			return
+		}
+		if !maxHash.IsEqual(blkHash) || blkHeight != i {
+			t.Errorf(
+				"NewestSha returned %v (%v), expected %v (%v)",
+				blkHeight,
+				maxHash,
+				i,
+				blkHash,
+			)
```

Comment:
Using multiple lines for arguments like this doesn't match the rest of the codebase.